tests/test_client.py

```diff
@@ -503,18 +504,26 @@ async def test_get_sensors(
     assert "sensors {" in query
 
 
+@pytest.mark.parametrize("scenario", [pytest.param("present"), pytest.param("missing")])
```

Comment:
I think the way pytest intends you to do this is to use [Indirect parametrization](https://docs.pytest.org/en/7.1.x/example/parametrize.html#indirect-parametrization). Something like this:

```
@fixture
def flow_summary_json(param):
  if param:
    yield {"valid": "json"}
  else:
    yield None


@pytest.mark.parametrize("flow_summary_json", (True, False), indirect=True)
def test_get_water_flow_summary(..., flow_summary_json):
  ...
  mock_session.execute.return_value = {"controller": {"sensors": [flow_sensor_json | {"flowSummary": flow_summary_json}]}}
```

Reply:
Didn't know that. Thanks. Fixed.